azerothcore/azerothcore-wotlk
# feat(Common/IPLocation): Replace ip2nation by ip2location

## Download
- https://download.ip2location.com/lite/ `IP2LOCATION-LITE-DB1.CSV`
- Full URL https://download.ip2location.com/lite/IP2LOCATION-LITE-DB1.CSV.ZIP

## Changes Proposed:
- https://github.com/TrinityCore/TrinityCore/commit/2fe4ab94c52ad2fa90189ecd9c053da49f06561e

## Issues Addressed:
- Close #5631

## Tests Performed:
- In-game

## How to Test the Changes:
- Download and set in config files `IPLocationFile`
- Set debug or trace level logger `root`
- Enter in-game
- Check `authserver` console, like this:
```
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to ip
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to country
```
- `.acc l c on`
- Try to re-enter in-game
- Check `authserver` console, like this:
```
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to ip
[AuthChallenge] Account 'WINFIDONARLEYAN' is locked to country: '-' Player country is '-'
```
- `.acc l c off`
- Try to re-enter in-game
- Check `authserver` console, like this:
```
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to ip
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to country
```

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
